assignment/index.html

```diff
@@ -8,6 +8,10 @@
   <body>
     <h1>Unit 5, Part 4 - Music Collection</h1>
     <p>See console for output.</p>
-    <!-- Create an unordered list with any questions you have -->
+    <br>
+    <ul>
+      <li>What are some reasons a return statement is marked as 'illegal return statement' in the console browser?</li>
+      <li>When using the  => (arrow) in a function, is it still good practice to include a 'return statement'? Research shows its not 'always' required.</li>
```

Comment:
The first thing to remember is that you can only leave out the return statement when your function only has one "expression" and you want to return the results of that expression. Now a single expression can get quite long, particularly when you start chaining array methods, so when you are first starting out it can be tough to spot what is a "single expression" and what is not. A good rule of thumb is that if you were to write the function _with_ a return statement, and the `return` is the first thing you write in your function, then you _could_ rewrite it as an arrow function _without_ the return.

```js
function singleExpression() {
  return /* expression goes here */;
}
```

Now, once we get past when you _can_ use an arrow function with an "implicit" return, when you _should_ is mostly a matter of personal preference. Different engineers will have different opinions, and your computer won't care either way. Hopefully whatever team you are working on has well-defined rules that are easy to follow.

That said, if you want to know _my_ personal preference, it is this:

1. DO use arrow functions with implicit returns when you have a function that is passed to another function (i.e. a callback) and it is clearly readable as a single line of code.

```js
let doubled = numbers.map(num => num * 2);
```

2. Do NOT use implicit returns when a function is more clearly written as multiple lines of code, even if it is technically only a single expression. There are ways to break up a single expression over multiple lines and still not use a return, but I do not like them. Just use an explicit return if your function is that long.

```js
let formatted = titles.map((title) => {
  return title
    .trim()
    .split(' ')
    .map(word => word[0].toUpperCase() + word.slice(1))
    .join(' ');
});
```

3. For named functions, I MAY use an implicit return for functions that can be cleanly written in a single line, but I more often use explicit returns or even full function declarations, even when the function in question is quite short.

```js
const double = num => num * 2;

const square = (number) => {
  return number ** 2;
};

function increment(number) {
  return number + 1;
}
```

4. DO follow a consistent pattern within a given team or project. Hopefully I have a linter or a formatter which enforces this pattern in an automated way, but the linters/formatters never catch everything anyway. A certain amount is always going to be on me to make sure my project has one consistent way of doing things (even if I switch it up a bit on the next project).